src/Enzo/enzo_EnzoConfig.cpp

```diff
@@ -340,10 +341,7 @@ void EnzoConfig::pup (PUP::er &p)
       method_grackle_chemistry->use_grackle = method_grackle_use_grackle;
 
       if(method_grackle_chemistry->use_grackle){
-        if (set_default_chemistry_parameters(method_grackle_chemistry) == ENZO_FAIL){
-          ERROR("EnzoMethodConfig::pup()",
-                "Error in Grackle set_default_chemistry_parameters");
-        }
+        *method_grackle_chemistry = _set_default_chemistry_parameters();
```

Comment:
Looking at this code again, I think this should probably be changed to 

```
method_grackle_chemistry = new chemistry_data;
if (method_grackle_use_grackle){
  *method_grackle_chemistry = _set_default_chemsitry_parameters();
}
method_grackle_chemistry->use_grackle = method_grackle_use_grackle;
```
Since `_set_default_chemistry_parameters()` resets `use_grackle` to `FALSE`. This issue is there in the previous version too, so I'm not too sure why this didn't cause problems.

Reply:
This is done correctly in `EnzoConfig::read`

Reply:
Thanks for catching this! I don't think this code has ever actually been tested because I believe that `EnzoConfig::pup` is only used for restarts. However, I think this wasn't actually a problem since the subsequent `p | *method_grackle_chemistry;` overwrites all of the attributes anyway. 

Now that I'm thinking about it, I the `set_default_chemistry_parameters(method_grackle_chemistry);` in the earlier version of the `pup` method was only necessary to set the value of the `grackle_data` global variable.  This means it isn't necessary to call `_set_default_chemsitry_parameters();`.

While we are on the topic, do you remember if there is a reason why we explicitly compute the `is_null` variable? It seems redundant with the value of `!method_grackle_chemistry`.

Reply:
Ah ok. That makes sense. So we can safely get rid of the call to `_set_default_chemistry_parameters()` here then? 

I vaguely remember the `is_null` variable being either from @jobordner or something I had copied based on similar code elsewhere. But I can't find the latter now and looking at the log, this bit was put in by @jobordner . Looking at the Charm++ documentation this is the way this should be done if the pointer is potentially NULL (https://charm.readthedocs.io/en/latest/charm++/manual.html#null-object-pointer)